### PR TITLE
Savon v2

### PR DESCRIPTION
**Pull request by ERCubed**
The initial intent of this pull was to update the gem to ensure that it works with the latest version of Savon. In doing so and adding in the WSDL which is particular to our project and account, I also came across several issues regarding Zuora functionality post WSDL version 46. I've made some changes and documented where the issues are (and admit that it needs to be handled better) and wanted to start a dialog with you over these changes and where else things may have changed. 

All tests but one pass right now. The failing one is:
rspec ./spec/subscribe_integration_spec.rb:128 # Subscribe Can successfully subscribe and generate an invoice
and that issue is "Net::ReadTimeout", which is something I'm guessing is due to the settings in our Zuora account and not the actual code itself. 

I doubt this is merge-ready, but I would welcome any discussion on making it merge-ready.

Regards,
-Eric

### CODE REVIEW

File: spec/subscribe_integration_spec.rb [hunk 1]
```diff
@@ -9,6 +9,8 @@
       @product = Z::Product.where(:name => "Awesome Product").first ||
         Z::Product.create!(
           :name => "Awesome Product",
+          :features__c => "Now with even more awesome features!",
```
Comment (anfleene) [inline spec/subscribe_integration_spec.rb]: I like the Idea of testing custom fields but I don't think we should do this. It makes it harder for others to run the tests against their accounts.

**Comment by anfleene** [comment]
First off, thanks for your contribution!
I'm a little concerned about the changes caused by updating the WSDL. I see this PR with 2 distinct goals.

1. Get AZ on a current version of Savon. 
2. Update the WSDL so we stop using such an old version of the Zuora API.

Both of these things are great and I'd like to accomplish them. The first one seems less risky though.
What if we went to Savon 2 in a single PR and then worked towards updating the WSDL in another?

I'd rather be as backward compatible for the previous version of the api as possible. I know it won't be easy/possible but I'd like to try. At least by isolating them into 2 different PRs we can attack them individually, my gut tells me upgrading Savon my have some of it's own issues and I'd rather not confuse them with the WSDL upgrade.

Thanks again!
Andy

File: spec/subscribe_integration_spec.rb (hunk 1 above)
Comment (ERCubed) [inline spec/subscribe_integration_spec.rb]: I'm fine with removing them in the end. I just needed to add them to work with our specific Zuora settings. I'll make a note to remove them before the next pull request.

**Comment by ERCubed** [comment]
I started off with the intent to just upgrade to the latest version of Savon. When I started to try and get things to work with our WSDL was when I ended up with this mishmash pull request. 

I agree that splitting it into two different sets of tasks is a wise idea.  Feel free to close this one and I'll do a fresh pull and request based off of what I've done for each Savon and the WSDL issues. 

-Eric

**Comment by anfleene** [comment]
Awesome, will do thanks.

**Pull request closed without merging**